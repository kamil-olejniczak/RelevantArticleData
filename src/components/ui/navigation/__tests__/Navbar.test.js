import renderer from 'react-test-renderer';
import * as React from "react";
import withNavbar from "../Navbar";
import {shallow} from 'enzyme';
import {Link, MemoryRouter} from 'react-router-dom';

const fakeFunction = jest.fn();
const FakeComponent = () => (
    <div>Hello</div>
);

describe('Navbar tests', () => {
    const Navbar = withNavbar(FakeComponent);
    const navbarWithFunction = (<Navbar clearArticle={fakeFunction}/>);
    let wrapper;

    it('renders correctly', () => {
        const component = renderer.create(
            <MemoryRouter>
                {navbarWithFunction}
            </MemoryRouter>
        ).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should render li tag to clear article data when clearArticle method is passed as prop', () => {
        wrapper = shallow(navbarWithFunction);

        expect(wrapper.find('li')).toHaveLength(2);
    });

    it('should render Link to "/" when clearArticle method is not provided', () => {
        wrapper = shallow(<Navbar/>);

        expect(wrapper.find(Link)).toHaveLength(2);
    });

    it('should spread props on wrapped component', () => {
        const propName = 'fakeProp';
        const expectedProp = {[propName]: '123456'};
        wrapper = shallow(<Navbar {...expectedProp}/>);

        expect(wrapper.find(FakeComponent).prop(propName)).toContain(expectedProp[propName]);
    });
});