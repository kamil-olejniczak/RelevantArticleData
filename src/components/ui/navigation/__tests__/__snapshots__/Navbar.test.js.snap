// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests renders correctly 1`] = `
<div>
  <ul
    className="Navbar"
  >
    <li
      onClick={[Function]}
    >
      Search for an article
    </li>
    <a
      href="/recentlyVisited"
      onClick={[Function]}
    >
      Your articles
    </a>
    <li>
      <a
        href="http://github.com/kamil-olejniczak"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </li>
  </ul>
  <div>
    Hello
  </div>
</div>
`;
